exercises/async_loops/solution/solution.js

```diff
@@ -1,12 +1,8 @@
-function loadUsers(userIds, load, done) {
-  var completed = 0
-  var users = []
-  userIds.forEach(function(id, index) {
-    load(id, function(user) {
-      users[index] = user
-      if (++completed === userIds.length) return done(users)
-    })
-  })
-}
+const Promise = require('bluebird');
 
-module.exports = loadUsers
+module.exports = function loadUsers(userIds, load, done) {
+  load = Promise.promisify(load); // could be eliminated
+  return Promise.resolve(userIds)
+    .map(id => load(id))
+    .then(done);
```

Comment:
By calling `done` with the Promise chain's native `.then` get a more backwards compatible pattern as a bonus.